extensions/amp-carousel/0.2/amp-carousel.js

```diff
@@ -122,6 +122,11 @@ class AmpCarousel extends AMP.BaseElement {
     return isLayoutSizeDefined(layout);
   }
 
+  /** @override */
+  prerenderAllowed() {
```

Comment:
This extension uses `triggerAnalyticsEvent`, which doesn't seem to be guarded by `viewer.isVisible` at all. We either need to guard it in this extension, or we (preferably) need to add it to `triggerAnalyticsEvent`.

/cc @zhouyx

Reply:
We should probably file an issue for this. I don't really see the problem as long as amp-analytics itself guards sending server pings by visibility (which I hope is the case)

Reply:
https://github.com/ampproject/amphtml/issues/28244

Reply:
Thanks! I think we can submit this PR as amp-analytics itself doesn't do bad things early.

Reply:
Confirmed in #28244. Approved.